src/main/java/io/strimzi/utils/LogicalKafkaVersionEntity.java

```diff
@@ -0,0 +1,207 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.utils;
+
+import com.google.gson.stream.JsonReader;
+import com.google.gson.stream.JsonToken;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.URL;
+import java.net.URLConnection;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ *  A logical Kafka version
+ */
+public class LogicalKafkaVersionEntity {
+
+    private static final Logger LOGGER = LogManager.getLogger(LogicalKafkaVersionEntity.class);
+
+    private static final Pattern STRIMZI_TEST_CONTAINER_IMAGE_WITHOUT_KAFKA_VERSION = Pattern.compile("^test-container:(\\d+\\.\\d+\\.\\d+|latest)-kafka-.*$");
+    private static final String KAFKA_VERSIONS_URL_JSON = "https://raw.githubusercontent.com/strimzi/test-container-images/main/kafka_versions.yaml"; // this would be replaced by test-container-images url where json will be stored
```

Comment:
I forgot what we discussed before. Is this pulled every time the container is used? Do we have some _offline_ fallback?

Reply:
If you run tests via `mvn` or `IDE` this is fetched only once but obviously when you run it multiple times it will fetch as many times as you execute it.